pkg/util/util_test.go

```diff
@@ -83,6 +83,7 @@ func TestGetCertPool(t *testing.T) {
 	certPool, err := GetCertPool([]string{certFile1.Name(), certFile2.Name()})
 	assert.NoError(t, err)
 
+	// nolint:staticcheck // Ignoring certPool.Subjects is deprecated error.
```

Comment:
Having reviewed what is happening, the appropriate thing to do, IMO,  would be to rewrite this test to load the certificates from the file and then attempt to verify them, if they can be verified using the `certPool`, we know it loaded them correctly

We can use `os.ReadFile` and then `x509.ParseCertificate`, then `certificate.Verify` I believe

Reply:
Thank you for you reviewing.
Your suggestion sounds better.

So, this test case should be like this, right?

Step1. Prepare (1) root ca (2) cert1 signed by the root ca (3) cert2 signed by the root ca
Step2. Create a CertPool with the root ca
Step3. Read cert1 and cert2 with `os.ReadFile` and `x509.ParseCertificate`
Step4. Verify cert1 and cert2 with `certificate.Verify` using the CertPool

Reply:
If you can make that work, I'm happy with that, sounds great!

Reply:
@JoelSpeed 

I did this work and it worked alright.

```
##### Check go version #####
ubuntu@wsl$ go version
go version go1.18 linux/amd64

##### Before fix #####
ubuntu@wsl$ git checkout tags/v7.3.0
HEAD is now at db74661e Merge pull request #1665 from oauth2-proxy/release-7.3.0
ubuntu@wsl$
ubuntu@wsl$ make lint
GO111MODULE=on golangci-lint run
WARN [runner] The linter 'scopelint' is deprecated (since v1.39.0) due to: The repository of the linter has been deprecated by the owner.  Replaced by exportloopref.
WARN [runner] The linter 'golint' is deprecated (since v1.41.0) due to: The repository of the linter has been archived by the owner.  Replaced by revive.
pkg/util/util_test.go:86:10: SA1019: certPool.Subjects is deprecated: if s was returned by SystemCertPool, Subjects will not include the system roots. (staticcheck)
        subj := certPool.Subjects()
                ^
make: *** [Makefile:34: lint] Error 1
ubuntu@wsl$

##### After fix ######
ubuntu@wsl$ git checkout  fix/deprecated-lint-issue
Switched to branch 'fix/deprecated-lint-issue'
ubuntu@wsl$
ubuntu@wsl$ make lint
GO111MODULE=on golangci-lint run
WARN [runner] The linter 'scopelint' is deprecated (since v1.39.0) due to: The repository of the linter has been deprecated by the owner.  Replaced by exportloopref.
WARN [runner] The linter 'golint' is deprecated (since v1.41.0) due to: The repository of the linter has been archived by the owner.  Replaced by revive.
ubuntu@wsl$
```